app/controllers/visitors_details_controller.rb

```diff
@@ -1,22 +1,54 @@
 class VisitorsDetailsController < ApplicationController
   include CookieGuard
   include SessionGuard::OnEditAndUpdate
-  include VisitorsManipulator
+  include TrimParams
+
   before_action :ensure_visit_integrity, only: [:edit, :update]
+  before_action :logstasher_add_visit_id_from_session
 
   def edit
+    visit.visitors = [Visitor.new] if visit.visitors.empty?
     @collect_phone_number = true
   end
 
-  def this_path
-    edit_visitors_details_path
+  def update
+    m = params[:next].match(/remove-(\d)/)
```

Comment:
Ah yes, I see what you mean. There's one way that uses `next; 'remove-1'` etc., which doesn't seem to be exercised anywhere except in one spec (and is thus presumably just junk code), and another that sets a `_destroy` attribute – that presumably doesn't work without JavaScript.